Sources/Sentry/SentryCrashIntegrationSessionHandler.m

```diff
@@ -36,31 +36,50 @@ - (instancetype)initWithCrashWrapper:(SentryCrashWrapper *)crashWrapper
     return self;
 }
 
-- (void)endCurrentSessionAsCrashedWhenCrashOrWatchdogTermination
+- (void)endCurrentSessionIfRequired
 {
+    SentryFileManager *fileManager = [[[SentrySDK currentHub] getClient] fileManager];
```

Comment:
`l`: Maybe some debug logs could be useful in this method as there are multiple conditions